services/app-api/src/postgres/contractAndRates/updateDraftContract.ts

```diff
@@ -0,0 +1,62 @@
+import { findContractWithHistory } from './findContractWithHistory'
+import { NotFoundError } from '../postgresErrors'
+import type { PrismaClient } from '@prisma/client'
+import type {
+    ContractType,
+    ContractFormEditableType,
+} from '../../domain-models/contractAndRates'
+import { prismaUpdateContractFormDataFromDomain } from './prismaContractRateAdaptors'
+
+type UpdateContractArgsType = {
+    contractID: string
+    formData: ContractFormEditableType
+}
+
+// Update the given draft
+async function updateDraftContract(
+    client: PrismaClient,
+    args: UpdateContractArgsType
+): Promise<ContractType | NotFoundError | Error> {
+    const { contractID, formData } = args
+
+    try {
+        return await client.$transaction(async (tx) => {
+            // Given all the Contracts associated with this draft, find the most recent submitted
```

Comment:
this comment doesn't quite seem right. We're looking for an unsubmitted contract rev